src/System.Security.Cryptography.Hashing/src/System/Security/Cryptography/KeyedHashAlgorithm.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+
+namespace System.Security.Cryptography
+{
+    public abstract class KeyedHashAlgorithm : HashAlgorithm
+    {
+        protected KeyedHashAlgorithm()
+        {
+        }
+
+        public virtual byte[] Key
+        {
+            get
+            {
+                return _key.CloneByteArray();
+            }
+
+            set
+            {
+                _key = value.CloneByteArray();
```

Comment:
Yeah, the only time that the key could meaningfully change while a hash is in progress with this contract is a threaded assignment during `CalculateHash(Stream)`.  That looks like it will result in calculating the hash as a skip of however many bytes were already read (from the current implementations).  But, the usual 'this isn't multi-threading-hardened' caveats apply.